chunky/src/java/se/llbit/chunky/renderer/CachedObjectProvider.java

```diff
@@ -0,0 +1,78 @@
+package se.llbit.chunky.renderer;
+
+import se.llbit.math.Ray;
+import se.llbit.math.Vector3;
+
+import java.util.ArrayList;
+import java.util.function.Supplier;
+
```

Comment:
Could you add a quick JavaDocs to describe this class and when to use it?
Btw. this is an object pool right? Maybe add pool somewhere in the name, because this describes the design pattern (https://en.wikipedia.org/wiki/Object_pool_pattern).